providers/flagsmith/src/test/java/dev.openfeature.contrib.providers.flagsmith/FlagsmithProviderTest.java

```diff
@@ -0,0 +1,340 @@
+            }
+            return new MockResponse().setResponseCode(404);
+        }
+    };
+
+    private static Stream<Arguments> provideKeysForFlagResolution() {
+        return Stream.of(
+            Arguments.of("true_key", "getBooleanEvaluation", Boolean.class, "true"),
+            Arguments.of("false_key", "getBooleanEvaluation", Boolean.class, "false"),
+            Arguments.of("string_key", "getStringEvaluation", String.class, "string_value"),
+            Arguments.of("int_key", "getIntegerEvaluation", Integer.class, "1"),
+            Arguments.of("double_key", "getDoubleEvaluation", Double.class, "3.141"),
+            Arguments.of("object_key", "getObjectEvaluation", Value.class, "{\"name\":\"json\"}")
+        );
+    }
+
+    private static Stream<Arguments> provideDisabledKeysForFlagResolution() {
+        return Stream.of(
+            Arguments.of("true_key_disabled", "getBooleanEvaluation", Boolean.class, "false"),
+            Arguments.of("false_key_disabled", "getBooleanEvaluation", Boolean.class, "true"),
+            Arguments
+                .of("string_key_disabled", "getStringEvaluation", String.class, "no_string_value"),
+            Arguments.of("int_key_disabled", "getIntegerEvaluation", Integer.class, "2"),
+            Arguments.of("double_key_disabled", "getDoubleEvaluation", Double.class, "1.47"),
+            Arguments
+                .of("object_key_disabled", "getObjectEvaluation", Value.class, "{\"name\":\"not_json\"}")
+        );
+    }
+
+    private static Stream<Arguments> provideBooleanKeysForEnabledFlagResolution() {
+        return Stream.of(
+            Arguments.of("true_key", "true", null),
+            Arguments.of("false_key", "true", null),
+            Arguments.of("true_key_disabled", "false", Reason.DISABLED.name()),
+            Arguments.of("false_key_disabled", "false", Reason.DISABLED.name())
+        );
+    }
+
+    private static Stream<Arguments> invalidOptions() {
+        return Stream.of(
+            null,
+            Arguments.of(FlagsmithProviderOptions.builder().build()),
+            Arguments.of(FlagsmithProviderOptions.builder().apiKey("").build())
+        );
+    }
+
+    private static Stream<Arguments> invalidCacheOptions() {
+        return Stream.of(
+            Arguments
+                .of(FlagsmithProviderOptions.builder().apiKey("API_KEY").expireCacheAfterAccess(1)
+                                            .build()),
+            Arguments
+                .of(FlagsmithProviderOptions.builder().apiKey("API_KEY").maxCacheSize(1).build()),
+            Arguments
+                .of(FlagsmithProviderOptions.builder().apiKey("API_KEY").expireCacheAfterWrite(1)
+                                            .build()),
+            Arguments.of(FlagsmithProviderOptions.builder().apiKey("API_KEY").recordCacheStats(true)
+                                                 .build())
+        );
+    }
+
+    @BeforeEach
```

Comment:
Seems like this BeforeEach isn't _actually_ used in all the tests - can we perhaps use test groups or something for this?